plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/MySqlClient.java

```diff
@@ -496,7 +497,7 @@ public boolean isNull(ResultSet resultSet, int columnIndex)
                     throws SQLException
             {
                 // super calls ResultSet#getObject(), which for TIMESTAMP type returns java.sql.Timestamp, for which the conversion can fail if the value isn't a valid instant in server's time zone.
-                resultSet.getObject(columnIndex, String.class);
+                resultSet.getObject(columnIndex, LocalDateTime.class);
```

Comment:
Is the above comment still valid?

Reply:
`ResultSet#getObject()` also solves this issue, but since it returns `java.sql.Timestamp` for both Mysql `datetime` and `timestamp` I prefer to set the exact type used by the `StandardColumnMappings#timestampReadFunction` and not to fail with any other corner case values.

I think that if we keep the override of `isNull()` than the comment is not relevant anyway.